src/snowflake/connector/auth/by_plugin.py

```diff
@@ -89,18 +59,39 @@ class AuthType(Enum):
 class AuthByPlugin(ABC):
     """External Authenticator interface."""
 
-    def __init__(self) -> None:
-        self._retry_ctx = AuthRetryCtx()
+    def __init__(
+        self,
+        timeout: int | None = None,
+        backoff_generator: Iterator | None = None,
+        **kwargs,
+    ) -> None:
         self.consent_cache_id_token = False
-        self._timeout: int = 120
+
+        max_retry_attempts = kwargs.get(
+            "max_retry_attempts",
+            int(getenv("MAX_CON_RETRY_ATTEMPTS", DEFAULT_MAX_CON_RETRY_ATTEMPTS)),
+        )
+        self._retry_ctx = TimeoutBackoffCtx(
+            timeout=timeout if timeout is not None else DEFAULT_AUTH_CLASS_TIMEOUT,
+            max_retry_attempts=max_retry_attempts,
+            backoff_generator=backoff_generator,
+        )
```

Comment:
nit: I think the code can be further simplified to:
```suggestion
        self._retry_ctx = TimeoutBackoffCtx(
            timeout=timeout if timeout is not None else DEFAULT_AUTH_CLASS_TIMEOUT,
            max_retry_attempts=kwargs.get(
            "max_retry_attempts",
            int(getenv("MAX_CON_RETRY_ATTEMPTS", DEFAULT_MAX_CON_RETRY_ATTEMPTS)),
            ),
            backoff_generator=backoff_generator,
        )
```